smart_contracts/scripts/genesis/config.ts

```diff
@@ -24,130 +27,209 @@ export interface Config {
   legacyMapping: LegacyMappingRegistryConfig
 }
 
-const contractsAddresses = {
-  ethereumDidRegistry: '0x0000000000000000000000000000000000003333',
-  credentialDefinitionRegistry: '0x0000000000000000000000000000000000004444',
-  schemas: '0x0000000000000000000000000000000000005555',
-  roles: '0x0000000000000000000000000000000000006666',
-  validators: '0x0000000000000000000000000000000000007777',
-  accountControl: '0x0000000000000000000000000000000000008888',
-  upgradeControl: '0x0000000000000000000000000000000000009999',
-  legacyMappingRegistry: '0x0000000000000000000000000000000000019999',
+const contractsAddresses: { [key: string]: string } = {}
+
+export function readContractsConfig() {
+  const configPath = resolve('..', inFile)
+
+  const data = readFileSync(configPath, 'utf8')
+  const parsed_data = JSON.parse(data)
+  const contracts = parsed_data.contracts
+
+  for (const key of Object.keys(contracts)) {
+    contractsAddresses[key] = contracts[key].address
+  }
+}
+
+function getContractAddress(contractName: string): string {
+  if (Object.keys(contractsAddresses).length === 0) {
+    throw new Error("the 'readContractsConfig()' function must be called")
+  }
+
+  if (contractsAddresses[contractName] === undefined) {
+    throw new Error(`contract '${contractName}' not found`)
+  }
+
+  return contractsAddresses[contractName]
 }
 
 export const config: Config = {
```

Comment:
let's avoid using of global variable `contractsAddresses` and convert `config` into function reading config file and preparing data like now.  
Build config in the main. 
Pass sub-config into corresponding contract function.